packages/react/src/components/ComboBox/ComboBox-test.js

```diff
@@ -1432,4 +1432,20 @@ describe('ComboBox', () => {
       expect(findInputNode()).toHaveDisplayValue('Option 2');
     });
   });
+
+  it('passes inputProps to the input element', () => {
+    render(
+      <ComboBox
+        id="test-combo"
+        items={[{ label: 'Item 1' }]}
+        itemToString={(item) => item?.label || ''}
+        inputProps={{ maxLength: 10, placeholder: 'Type here' }}
+      />
+    );
+
+    const input = screen.getByPlaceholderText('Type here');
+    expect(input).toBeInTheDocument();
+    expect(input).toHaveAttribute('maxLength', '10');
+    expect(input).toHaveAttribute('placeholder', 'Type here');
```

Comment:
On https://github.com/carbon-design-system/carbon/pull/19461#discussion_r2106272864, I meant something more like the following where everything that is there is asserted:

```suggestion
    const attributes = Array.from(input.attributes).reduce(
      (acc, { name, value }) => ({ ...acc, [name]: value }),
      {}
    );

    expect(input).toBeInTheDocument();
    expect(attributes).toEqual({
      'aria-activedescendant': '',
      'aria-autocomplete': 'list',
      'aria-controls': 'downshift-:r7r:-menu',
      'aria-expanded': 'false',
      'aria-haspopup': 'listbox',
      'aria-label': 'Choose an item',
      autocomplete: 'off',
      class: 'cds--text-input cds--text-input--empty',
      id: 'test-combo',
      maxlength: '10',
      placeholder: 'Type here',
      role: 'combobox',
      tabindex: '0',
      type: 'text',
      value: '',
    });
```